arangod/Agency/AgencyComm.cpp

```diff
@@ -692,19 +692,44 @@ void AgencyCommManager::addEndpoint(std::string const& endpoint) {
   if (iter == _endpoints.end()) {
     LOG_TOPIC(DEBUG, Logger::AGENCYCOMM) << "using agency endpoint '"
                                          << normalized << "'";
-
     _endpoints.emplace_back(normalized);
   }
 }
 
-void AgencyCommManager::removeEndpoint(std::string const& endpoint) {
+void AgencyCommManager::updateEndpoints(std::vector<std::string> const& newEndpoints) {
   MUTEX_LOCKER(locker, _lock);
+  
+  std::set<std::string> currentSet;
+  auto iter = _endpoints.begin();
```

Comment:
could be simplified to a range-based for loop, or even `std::set::insert` with a range (`currentSet.insert(_endpoints.begin(), _endpoints.end());`)